crates/test-programs/wasi-tests/src/bin/path_link.rs

```diff
@@ -68,30 +95,38 @@ unsafe fn check_rights(orig_fd: wasi::Fd, link_fd: wasi::Fd) {
     let link_filestat = wasi::fd_filestat_get(link_fd).expect("reading filestat of the link");
     filestats_assert_eq(orig_filestat, link_filestat);
 
+    // Disabled: fd_fdstat_get causes CI test failures on Windows
     // Compare Fdstats
     let orig_fdstat = wasi::fd_fdstat_get(orig_fd).expect("reading fdstat of the source");
     let link_fdstat = wasi::fd_fdstat_get(link_fd).expect("reading fdstat of the link");
     fdstats_assert_eq(orig_fdstat, link_fdstat);
 }
 
+// Use of WasiFile is needed for Windows, which will not remove the directory until all
```

Comment:
If we remove the RAII type, we should remove or update this comment.  Calling out that the `fd_close` must be called prior to unlinking on Windows probably isn't worth a comment, but I'm also good with updating this comment to reflect that.